### PR TITLE
next: add skygear url to tenant configuration

### PR DESCRIPTION
**Pull request by kaojohnny**
connects #755 

1. add `SKYGEAR_URL` to `TenantConfiguration`, this is used in auth gear to generate a request to record gear.
2. add `record_enabled` to `Plan` table, it helps gateway to know if it routes record request of an app to record gear.

### CODE REVIEW

**Comment by Steven-Chan** [comment]
Also add a configuration for auth gear to know which user profile implementation chosen?

**Comment by kaojohnny** [comment]
@Steven-Chan We already have 

```
tConfig := config.GetTenantConfig(r)
userProfileStore := tConfig.UserProfile.ImplName
```

if `userProfileStore == "record"`, then auth gear will use record gear as user profile store. Or do you consider another configuration?

**Comment by Steven-Chan** [comment]
omg, me so forgetful 😢.

**Comment by Steven-Chan** [comment]
Configuration are already grouped by feature, e.g. `ASSET_STORE_XXX`, `TOKEN_STORE_XXX`.

First of all, I think the url added by this PR should be under `USER_PROFILE_XXX` instead of a generic `SKYGEAR_URL`.
Second, to @rickmak, I think grouping by feature is already enough.

**Comment by kaojohnny** [comment]
@Steven-Chan I think `SKYGEAR_URL=http://localhost:3001` is a general configuration for all gears, in current use case (save user profile to record gear), it is used when `auth` gear sends a request to `record` gear, something like:

```
resp, err := goreq.Request{
	Method: "POST",
	Uri:    u.skygearURL + "record/save",
	Body:   body,
}.Do()
```

It is more like a base configuration for communication between gears. Whereas, `USER_PROFILE_RECORD_STORE_URL=http://localhost:3001/record/` is also good to me.

**Comment by Steven-Chan** [comment]
Actually, I prefer `USER_PROFILE_SKYGEAR_URL` over `USER_PROFILE_RECORD_STORE_URL`.

File: pkg/core/config/tenant.go [hunk 1]
```diff
@@ -10,14 +10,15 @@ import (
...
-	UserAudit       UserAuditConfiguration   `json:"USER_AUDIT" msg:"USER_AUDIT"`
+	DBConnectionStr           string                   `msg:"DATABASE_URL" envconfig:"DATABASE_URL" json:"DATABASE_URL"`
+	APIKey                    string                   `msg:"API_KEY" envconfig:"API_KEY" json:"API_KEY"`
+	MasterKey                 string                   `msg:"MASTER_KEY" envconfig:"MASTER_KEY" json:"MASTER_KEY"`
+	AppName                   string                   `msg:"APP_NAME" envconfig:"APP_NAME" json:"APP_NAME"`
+	CORSHost                  string                   `msg:"CORS_HOST" envconfig:"CORS_HOST" default:"*"`
+	TokenStore                TokenStoreConfiguration  `json:"TOKEN_STORE" msg:"TOKEN_STORE"`
+	UserProfile               UserProfileConfiguration `json:"USER_PROFILE" msg:"USER_PROFILE"`
+	UserAudit                 UserAuditConfiguration   `json:"USER_AUDIT" msg:"USER_AUDIT"`
+	UserProfileRecordStoreURL string                   `msg:"USER_PROFILE_RECORD_STORE_URL" envconfig:"USER_PROFILE_RECORD_STORE_URL" json:"USER_PROFILE_RECORD_STORE_URL"`
```
Comment (Steven-Chan) [inline pkg/core/config/tenant.go]: This config should be under `UserProfileConfiguration`.

For config that have more than one implementation, I think you can take skygear v1 asset store as reference.

https://github.com/SkygearIO/skygear-server/blob/master/pkg/server/skyconfig/config.go#L152
https://github.com/SkygearIO/skygear-server/blob/master/pkg/server/skyconfig/config.go#L441

---

`UserProfileSkygearURL` or `UserProfileRecordStoreURL`? @carmenlau @rickmak any comment?

Comment (kaojohnny) [reply pkg/core/config/tenant.go]: @Steven-Chan I updated it to 

```
type UserProfileConfiguration struct {
	ImplName     string `msg:"IMPLEMENTATION" envconfig:"USER_PROFILE_IMPL_NAME" json:"IMPLEMENTATION"`
	ImplStoreURL string `msg:"IMPL_STORE_URL" envconfig:"USER_PROFILE_IMPL_STORE_URL" json:"IMPL_STORE_URL"`
}
```

let me know if it is a proper name. Thanks.

**Pull request merged**